### ISSUE TITLE
Calendar: onChange/onBeforeChange and keyboard input do work strange

### ISSUE THREAD

**Issue by PhilippGrashoff** [fomantic/Fomantic-UI#1517]
# Bug Report
When entering a value via keyboard into the input field of a calendar, it gets magically turned into a date in onChange/onBeforeChange. Even typing only a single "1" results in the date 2001-01-01.

## Steps to reproduce
1. use this simple jsFiddle: https://jsfiddle.net/nyg4vph1/3/
2. type anything in the date input. You will see in the console that even on the first sign you type, the parameters passed to onChange represent a real date. This is what IMO the bug is: Some user input which is not a valid date gets magically turned into some date, even if you only type "1" in the empty input.
3. Due to this, I cant see a way to determine in onChange() if the user actually typed a date or just anything that gets converted into a date.

## Expected result
Do not magically convert any typed input into a date, only if its in the defined format.

##Why
Think of an application which submits the form in the onChange handler. You will want to check first if the date the user entered is a valid date, and only then proceed. This is not possible, as any input gets turned into some date. Think of someone buying a ticket for a movie, accidently types the date incomplete, and instead of being noticed about this the user goes on buying a ticket for a date he never selected.

## Version
2.8.5

**Comment by PhilippGrashoff** [comment]
But I do not want to disable the manual date input. Its very handy in case you have to enter the date which is for Example 3 Years in the future. Would need 36 clicks in the calendar.

**Comment by lubber-de** [comment]
>  you have to enter the date which is for Example 3 Years in the future. Would need 36 clicks in the calendar.

Do you know you can click on the Year to select the year much more faster? So its about 6-7 clicks rather than 36 :wink:
|||
|-|-|
|![datein3years](https://user-images.githubusercontent.com/18379884/84360709-d2680f80-abca-11ea-8a17-b2c12a600e92.gif)|![datein3years_2](https://user-images.githubusercontent.com/18379884/84360720-d5fb9680-abca-11ea-94e9-1be6d1eabbd0.gif)|

However, i do understand entering the date manually is probably much faster anyway

**Comment by PhilippGrashoff** [comment]
Hi, I know that the date can be clicked, but from experience with the average user, I am very sure 95% dont :)

**Comment by lubber-de** [comment]
Yes, that might be true 🤣 , i just wanted to clarify 36 clicks are not really needed technical speaken 😉 
So, back to the main topic: I initially suggestd to implment either a `minCharacter` setting or some kind of format check before a date is even tried to be parsed.
For the latter, we already have a feature request for general masked inputs #368 which might help here also
